web/frontend/src/components/slam.vue

```diff
@@ -95,6 +107,89 @@ const fetchSLAMPose = (name: string): Promise<commonApi.Pose> => {
   });
 };
 
+const executeMoveOnMap = async () => {
+  window.localStorage.setItem(operationID, operationID)
+  moveClick = !moveClick;
+
+  /*
+    * set request name
+    * here we set the name of the motion service the user is using
+  */
+  motionServiceReq.setName('builtin');
+
+  // set pose in frame
+  const destination = new commonApi.Pose();
+  const value = await fetchSLAMPose(props.name);
+  destination.setX(destinationMarker.x);
+  destination.setY(0);
+  destination.setZ(destinationMarker.z);
+  destination.setOX(value.getOX());
+  destination.setOY(value.getOY());
+  destination.setOZ(value.getOZ());
+  destination.setTheta(value.getTheta());
+  motionServiceReq.setDestination(destination);
+
+  // set SLAM resource name
+  const slamResourceName = new commonApi.ResourceName();
+  slamResourceName.setNamespace('rdk');
+  slamResourceName.setType('service');
+  slamResourceName.setSubtype('slam');
+  slamResourceName.setName(props.name);
+  motionServiceReq.setSlamServiceName(slamResourceName);
+
+  // set component name
+  const baseResources = filterResources(props.resources, 'rdk', 'component', 'base');
+  if (baseResources === undefined) {
+    toast.error('No base component detected.');
+    return;
+  }
+  const baseResourceName = new commonApi.ResourceName();
+  baseResourceName.setNamespace('rdk');
+  baseResourceName.setType('component');
+  baseResourceName.setSubtype('base');
+  baseResourceName.setName(baseResources[0]!.name);
+  motionServiceReq.setComponentName(baseResourceName);
+
+  // set extra as position-only constraint
+  motionServiceReq.setExtra(
+    Struct.fromJavaScript({
+      motion_profile: "position_only"
+    })
+  )
+
+  props.client.motionService.moveOnMap(
+    motionServiceReq,
+    new grpc.Metadata(),
+    (error: ServiceError | null, response: motionApi.MoveOnMapResponse | null) => {
+      if (error) {
+        window.localStorage.removeItem(operationID)
```

Comment:
If the user refreshes the page while the move command is in flight, won't that mean `window.localStorage.removeItem(operationID)` is never called & no move calls can happen until the user clears local storage (which is not intuitive or obvious)?

Reply:
yes, but in this case the user can press the stop button.